src/module_wrap.cc

```diff
@@ -502,6 +502,44 @@ void ModuleWrap::Resolve(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result.ToObject(env));
 }
 
+static MaybeLocal<Promise> ImportModuleDynamically(Local<Context> context,
+                                                   Local<String> referrer,
+                                                   Local<String> specifier) {
+  Isolate* iso = context->GetIsolate();
+  // TODO(jkrems): ensure proper context is used
+  Environment* env = Environment::GetCurrent(context);
+  HandleScope handle_scope(iso);
+
+  Local<Function> import_callback =
+    env->host_import_module_dynamically_callback();
+  Local<Value> import_args[] = {
+    Local<Value>(referrer),
+    Local<Value>(specifier)
+  };
+  MaybeLocal<Value> import_result = import_callback->Call(context,
+                                                          v8::Undefined(iso),
+                                                          2,
+                                                          import_args);
+  // TODO(jkrems): Actual error handling
+  return import_result.ToLocalChecked().As<Promise>();
+}
+
+void ModuleWrap::SetImportModuleDynamicallyCallback(
+    const FunctionCallbackInfo<Value>& args) {
+  Isolate* iso = args.GetIsolate();
+  Environment* env = Environment::GetCurrent(args);
+  EscapableHandleScope handle_scope(iso);
```

Comment:
Does the handle scope needs to be escapable?

Reply:
Don't think so, bad copy&paste. Thanks!